src/components/ciConfig/TargetPlatformSelector.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react'
+import { components } from 'react-select'
+import CreatableSelect from 'react-select/creatable'
+import { ReactComponent as WarningIcon } from '../../assets/icons/ic-warning.svg'
+import { DockerConfigOverrideKeys } from '../ciPipeline/types'
+import { TARGET_PLATFORM_LIST, tempMultiSelectStyles } from './CIConfig.utils'
+import { SelectorMessaging } from './ciConfigConstant'
+import { TargetPlatformSelectorType } from './types'
+
+const platformMenuList = (props): JSX.Element => {
+    return (
+        <components.MenuList {...props}>
+            <div className="cn-5 pl-12 pt-4 pb-4 dc__italic-font-style">{SelectorMessaging.TARGET_SELECTOR_MENU}</div>
+            {props.children}
+        </components.MenuList>
+    )
+}
+
+function TargetPlatformSelector({
+    allowOverride,
+    selectedTargetPlatforms,
+    setSelectedTargetPlatforms,
+    showCustomPlatformWarning,
+    setShowCustomPlatformWarning,
+    targetPlatformMap,
+    targetPlatform,
+    configOverrideView,
+    updateDockerConfigOverride,
+}: TargetPlatformSelectorType) {
+    const noMatchingPlatformOptions = (): string => {
```

Comment:
We should move it in helper file as this has been used on multiple places

Reply:
Done